dotnet/src/SemanticKernel.UnitTests/KernelTests.cs

```diff
@@ -152,45 +162,78 @@ public void ItAllowsToImportTheSamePluginMultipleTimes()
         kernel.ImportFunctions(new MyPlugin());
     }
 
-    [Fact]
-    public async Task ItUsesDefaultServiceWhenSpecifiedAsync()
+    [Theory]
+    [InlineData(null)]
+    [InlineData(true)]
+    [InlineData(false)]
+    public async Task ItUsesDefaultServiceWhenSpecifiedAsync(bool? useStream)
     {
         // Arrange
-        var mockTextCompletion1 = new Mock<ITextCompletion>();
-        var mockTextCompletion2 = new Mock<ITextCompletion>();
-        var mockCompletionResult = new Mock<ITextResult>();
+        var mockTextCompletion1 = new Mock<ITextStreamingCompletion>();
+        var mockTextCompletion2 = new Mock<ITextStreamingCompletion>();
+        var mockCompletionResult = new Mock<ITextStreamingResult>();
+
+        mockTextCompletion1.Setup(c => c.GetTextStreamingResultsAsync(It.IsAny<string>(), It.IsAny<AIRequestSettings>(), It.IsAny<CancellationToken>())).Returns(new[] { mockCompletionResult.Object }.ToAsyncEnumerable());
+        mockTextCompletion1.Setup(c => c.GetTextResultsAsync(It.IsAny<string>(), It.IsAny<AIRequestSettings>(), It.IsAny<CancellationToken>())).ReturnsAsync(new[] { mockCompletionResult.Object });
+
+        mockTextCompletion2.Setup(c => c.GetTextStreamingResultsAsync(It.IsAny<string>(), It.IsAny<AIRequestSettings>(), It.IsAny<CancellationToken>())).Returns(new[] { mockCompletionResult.Object }.ToAsyncEnumerable());
+        mockTextCompletion2.Setup(c => c.GetTextResultsAsync(It.IsAny<string>(), It.IsAny<AIRequestSettings>(), It.IsAny<CancellationToken>())).ReturnsAsync(new[] { mockCompletionResult.Object });
 
-        mockTextCompletion1.Setup(c => c.GetCompletionsAsync(It.IsAny<string>(), null, It.IsAny<CancellationToken>())).ReturnsAsync(new[] { mockCompletionResult.Object });
-        mockTextCompletion2.Setup(c => c.GetCompletionsAsync(It.IsAny<string>(), null, It.IsAny<CancellationToken>())).ReturnsAsync(new[] { mockCompletionResult.Object });
-        mockCompletionResult.Setup(cr => cr.GetCompletionAsync(It.IsAny<CancellationToken>())).ReturnsAsync("llmResult");
+        mockCompletionResult.Setup(cr => cr.GetTextStreamingAsync(It.IsAny<CancellationToken>())).Returns(new[] { "llmResult" }.ToAsyncEnumerable());
+        mockCompletionResult.Setup(cr => cr.GetTextAsync(It.IsAny<CancellationToken>())).ReturnsAsync("llmResult");
 
         var kernel = Kernel.Builder
             .WithAIService<ITextCompletion>("service1", mockTextCompletion1.Object, false)
             .WithAIService<ITextCompletion>("service2", mockTextCompletion2.Object, true)
             .Build();
 
         var templateConfig = new PromptTemplateConfig();
+        if (useStream is not null)
+        {
+            templateConfig.Completion = new() { Streaming = useStream };
```

Comment:
As mentioned above, name `Streaming` here doesn't show that it's boolean flag, that's why this test contains parameter `useStream`. I would prefer to use the same naming in settings and tests, like `UseStreaming` or `IsStreamingEnabled`.